clever/src/led.cpp

```diff
@@ -0,0 +1,300 @@
+/*
+ * High level control for the LED strip
+ * Indicate flight events with the LED strip
+ * Copyright (C) 2019 Copter Express Technologies
+ *
+ * Author: Oleg Kalachev <okalachev@gmail.com>
+ *
+ * Distributed under MIT License (available at https://opensource.org/licenses/MIT).
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ */
+
+#include <ros/ros.h>
+#include <string>
+#include <boost/algorithm/string.hpp>
+
+#include <clever/SetLEDEffect.h>
+#include <led_msgs/SetLEDs.h>
+#include <led_msgs/LEDState.h>
+#include <led_msgs/LEDStateArray.h>
+
+#include <sensor_msgs/BatteryState.h>
+#include <mavros_msgs/State.h>
+#include <rosgraph_msgs/Log.h>
+
+clever::SetLEDEffect::Request current_effect;
+int led_count;
+ros::Timer timer;
+ros::Time start_time;
+double blink_rate, blink_fast_rate, flash_delay, fade_period, rainbow_period;
+double low_battery_threshold;
+bool blink_state;
+led_msgs::SetLEDs set_leds;
+led_msgs::LEDStateArray state, start_state;
+ros::ServiceClient set_leds_srv;
+mavros_msgs::State mavros_state;
+int counter;
+
+inline void call_set_leds()
```

Comment:
No real reason to make this `inline` - please [consult the meaning of this keyword in C++](https://en.cppreference.com/w/cpp/language/inline).

Reply:
OK.